tests/unit/completion/test_completer.py

```diff
@@ -147,96 +143,156 @@ def _set_cmd_prompt(cmd, txt):
+                              config_stub, status_command_stub,
+                              completion_widget_stub):
     """Test that on_selection_changed modifies the cmd text properly.
 
     The | represents the current cursor position in the cmd prompt.
     If quick-complete is True and there is only 1 completion (count == 1),
     then we expect a space to be appended after the current word.
     """
-    config_stub.data['completion']['quick-complete'] = quick_complete
     model = unittest.mock.Mock()
     model.data = unittest.mock.Mock(return_value=newtxt)
-    model.count = unittest.mock.Mock(return_value=count)
     indexes = [unittest.mock.Mock()]
     selection = unittest.mock.Mock()
     selection.indexes = unittest.mock.Mock(return_value=indexes)
     completion_widget_stub.model.return_value = model
-    _set_cmd_prompt(status_command_stub, before)
-    # schedule_completion_update is needed to pick up the cursor position
-    completer_obj.schedule_completion_update()
+
+    def check(quick_complete, count, expected_txt, expected_pos):
+        config_stub.data['completion']['quick-complete'] = quick_complete
+        model.count = unittest.mock.Mock(return_value=count)
+        _set_cmd_prompt(status_command_stub, before)
+        completer_obj.on_selection_changed(selection)
+        model.data.assert_called_with(indexes[0])
+        assert status_command_stub.text() == expected_txt
+        assert status_command_stub.cursorPosition() == expected_pos
+
+    after_pos = after.index('|')
+    after_txt = after.replace('|', '')
+    check(False, 1, after_txt, after_pos)
+    check(True, 2, after_txt, after_pos)
+
+    # quick-completing a single item should move the cursor ahead by 1 and add
+    # a trailing space if at the end of the cmd string
+    after_pos += 1
+    if after_pos > len(after_txt):
+        after_txt += ' '
+    check(True, 1, after_txt, after_pos)
+
+
+def test_quickcomplete_flicker(status_command_stub, completer_obj,
+                               completion_widget_stub, config_stub):
+    """Validate fix for #1519: bookmark-load background highlighting quirk.
+
+    For commands like bookmark-load and open with maxsplit=0, a commandline
+    that looks like ':open someurl |' is considered to be completing the first
+    arg with pattern 'someurl ' (note trailing whitespace). As this matches the
+    one completion available, it keeps the completionmenu open.
+
+    This test validates that the completion model is not re-set after we
+    quick-complete an entry after maxsplit.
+    """
+    model = unittest.mock.Mock()
+    model.data = unittest.mock.Mock(return_value='http://example.com')
+    indexes = [unittest.mock.Mock()]
+    selection = unittest.mock.Mock()
+    selection.indexes = unittest.mock.Mock(return_value=indexes)
+    completion_widget_stub.model.return_value = model
+
+    config_stub.data['completion']['quick-complete'] = True
+    model.count = unittest.mock.Mock(return_value=1)
```

Comment:
This whole setup is kinda ugly and duplicated from the test above (which I also wrote :grimacing:). I'll see if there's a cleaner way to do it or at least use spec / lambdas. I was thinking it might be cleaner (implementation-wise as well) to have `on_selection_chaged` take a `string` rather than a `QModelIndex` but that might just push the complexity elsewhere.